### PR TITLE
Added yaml and dictionary conversion methods to BlockMap

### PR DESCRIPTION
**Pull request by kc611**
As titled, this PR makes the `from_yaml` method a staticmethod on `BlockMap`. Also adds a deserializer to convert `BlockMap` to yaml and a similar conversion in dictionary format as well.

### CODE REVIEW

File: numba_rvsdg/core/datastructures/block_map.py [hunk 1]
```diff
@@ -318,3 +320,56 @@ def join_tails_and_exits(self, tails: Set[Label], exits: Set[Label]):
...
+        graph_dict = {}
+        for key, value in block_map_graph.items():
+            curr_dict = {}
+            curr_dict["jt"] = [f"{i.index}" for i in value._jump_targets]
+            if value.backedges:
+                curr_dict["be"] = [f"{i.index}" for i in value.backedges]
+            graph_dict[str(key.index)] = curr_dict
+        return graph_dict
+
+def wrap_id(indices: Set[Label]):
```
Comment (esc) [inline numba_rvsdg/core/datastructures/block_map.py:374]: Do you think this could go into `test_utils.py` too?

**Review by esc — CHANGES_REQUESTED** [review]
Few things to address.

File: numba_rvsdg/core/datastructures/block_map.py (hunk 1 above)
Comment (kc611) [reply numba_rvsdg/core/datastructures/block_map.py:374]: No, it's being used by functions within `block_map.py`. 

Besides, it'll go away anyways while the planned library rewrites so we need not bother anyway.

File: numba_rvsdg/tests/test_blockmap.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+from numba_rvsdg.core.datastructures.block_map import BlockMap
+
+from numba_rvsdg.tests.test_utils import MapComparator
+
+
+class TestBlockMapConversion(MapComparator):
+
+    def test_yaml_conversion(self):
+        # Case # 1
```
[the remark this answers is not in the document]
Comment (kc611) [reply numba_rvsdg/tests/test_blockmap.py]: Corrected in [8433152](https://github.com/numba/numba-rvsdg/pull/19/commits/843315208debd03ee616aa73d4236adc51ff2189)

File: numba_rvsdg/core/datastructures/block_map.py [hunk 3]
```diff
@@ -318,3 +320,56 @@ def join_tails_and_exits(self, tails: Set[Label], exits: Set[Label]):
...
+        # Convert to yaml
+        block_map_graph = self.graph
+        yaml_string = """"""
+
+        for key, value in block_map_graph.items():
+            jump_targets = [f"{i.index}" for i in value._jump_targets]
+            jump_targets = str(jump_targets).replace("\'", "\"")
+            back_edges = [f"{i.index}" for i in value.backedges]
+            yaml_string += f"""
+    "{str(key.index)}":
```
[the remark this answers is not in the document]
Comment (kc611) [reply numba_rvsdg/core/datastructures/block_map.py]: Corrected in [8433152](https://github.com/numba/numba-rvsdg/pull/19/commits/843315208debd03ee616aa73d4236adc51ff2189)

**Review by esc — APPROVED** [review]

**Comment by esc** [comment]
ty

**Pull request merged**